lib/TH/THGeneral.c

```diff
@@ -137,10 +159,17 @@ void* THRealloc(void *ptr, long size)
   if(size < 0)
     THError("$ Torch: invalid memory size -- maybe an overflow?");
 
-  ptr = realloc(ptr, size);
-  if(!ptr)
+  void *newptr = realloc(ptr, size);
+
+  if(!newptr && torchGCFunction) {
+    torchGCFunction(torchGCData);
+    newptr = realloc(ptr, size);
+  }
+
+  if(!newptr)
     THError("$ Torch: not enough memory: you tried to reallocate %dGB. Buy new RAM!", size/1073741824);
```

Comment:
Well, at least in the standard (Lua) implementation of THError, it calls lua_error which is a longjmp so this function will never return.

I suppose if this function did return after calling THError, it would leak.